kube-runtime/src/events.rs

```diff
@@ -64,6 +76,36 @@ pub enum EventType {
     Warning,
 }
 
+/// [`ObjectReference`] with Hash and Eq implementations
+///
+/// [`ObjectReference`]: k8s_openapi::api::core::v1::ObjectReference
+#[derive(Clone, Debug, PartialEq)]
+pub struct Reference(ObjectReference);
+
+impl Eq for Reference {}
+
+impl Hash for Reference {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.api_version.hash(state);
+        self.0.kind.hash(state);
+        self.0.name.hash(state);
+        self.0.namespace.hash(state);
+        self.0.uid.hash(state);
+    }
+}
+
+/// Isomorphic key for caching similar events
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct EventKey {
```

Comment:
nit: isomorphic is a pretty overloaded term, and i don't know what that actually means in this context. maybe there's something else to use?
```suggestion
/// Cache key for event deduplication
#[derive(Clone, Debug, PartialEq, Eq, Hash)]
struct CacheKey {
```

Reply:
Fixed on 9e3032367661c345dc2c755b4703a7fa993fc18d.